timeseries/src/autogluon/timeseries/models/ensemble/array_based/regressor/tabular.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from typing_extensions import Self
+
+from autogluon.tabular import TabularPredictor
+
+from .abstract import EnsembleRegressor
+
+logger = logging.getLogger(__name__)
+
+
+class TabularEnsembleRegressor(EnsembleRegressor):
+    """TabularPredictor ensemble regressor using AutoGluon-Tabular as a single
+    quantile regressor for the target.
+    """
+
+    def __init__(
+        self,
+        quantile_levels: list[float],
+        tabular_hyperparameters: Optional[dict] = None,
+    ):
+        super().__init__()
+        self.quantile_levels = quantile_levels
+        self.tabular_hyperparameters = tabular_hyperparameters or {}
+        self.predictor: Optional[TabularPredictor] = None
+
+    def fit(
+        self,
+        base_model_mean_predictions: np.ndarray,
+        base_model_quantile_predictions: np.ndarray,
+        labels: np.ndarray,
+        **kwargs,
+    ) -> Self:
+        time_limit = kwargs.get("time_limit", None)
+
+        self.predictor = TabularPredictor(
+            label="target",
+            problem_type="quantile",
+            quantile_levels=self.quantile_levels,
+            verbosity=1,
+        )
+
+        # get features
+        df = self._get_feature_df(base_model_mean_predictions, base_model_quantile_predictions)
+
+        # get labels
+        num_windows, num_items, prediction_length = base_model_mean_predictions.shape[:3]
+        label_series = labels.reshape(num_windows * num_items * prediction_length)
+        df["target"] = label_series
+
+        self.predictor.fit(
+            df,
+            hyperparameters=self.tabular_hyperparameters,
+            time_limit=time_limit,  # type: ignore
+        )
+
+        return self
+
+    def predict(
+        self,
+        base_model_mean_predictions: np.ndarray,
+        base_model_quantile_predictions: np.ndarray,
+    ) -> tuple[np.ndarray, np.ndarray]:
+        if self.predictor is None:
+            raise ValueError("Model must be fitted before prediction")
+
+        df = self._get_feature_df(base_model_mean_predictions, base_model_quantile_predictions)
+
+        pred = self.predictor.predict(df)
```

Comment:
If you plan to call predict multiple times, you may want to persist the models into memory with `self.predictor.persist()` for faster throughput (be aware of memory usage though)

Reply:
I don't think we are in this case. Thanks!